src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Map/GenerateOfflineMapWithOverrides/GenerateOfflineMapWithOverrides.xaml.cs

```diff
@@ -0,0 +1,428 @@
+
+            // Use that key to get the generate options for the layer.
+            GenerateGeodatabaseParameters generateParams = overrides.GenerateGeodatabaseParameters[layerKey];
+
+            // Return the layer options.
+            return generateParams.LayerOptions;
+        }
+
+        private void ExcludeLayerByName(string layerName, GenerateOfflineMapParameterOverrides overrides)
+        {
+            // Get the feature layer with the specified name.
+            FeatureLayer targetLayer = GetLayerByName(layerName);
+
+            // Get the layer's ID.
+            long targetLayerId = GetServiceLayerId(targetLayer);
+
+            // Create a layer key for the selected layer.
+            OfflineMapParametersKey layerKey = new OfflineMapParametersKey(targetLayer);
+
+            // Get the parameters for the layer.
+            GenerateGeodatabaseParameters generateParams = overrides.GenerateGeodatabaseParameters[layerKey];
+
+            // Get the layer options for the layer.
+            IList<GenerateLayerOption> layerOptions = generateParams.LayerOptions;
+
+            // Find the layer option matching the ID.
+            GenerateLayerOption targetLayerOption = layerOptions.First(layer => layer.LayerId == targetLayerId);
+
+            // Remove the layer option.
+            layerOptions.Remove(targetLayerOption);
+        }
+
+        private FeatureLayer GetLayerByName(string layerName)
+        {
+            // Get the first map in the operational layers collection that is a feature layer with name matching layerName
+            return MyMapView.Map.OperationalLayers.OfType<FeatureLayer>().First(layer => layer.Name == layerName);
+        }
+
+        private long GetServiceLayerId(FeatureLayer layer)
+        {
+            // Find the service feature table for the layer; this assumes the layer is backed by a service feature table.
+            ServiceFeatureTable serviceTable = (ServiceFeatureTable) layer.FeatureTable;
+
+            // Return the layer ID.
+            return serviceTable.LayerInfo.ServiceLayerId;
+        }
+
+        #endregion overrides
+
+        // Show changes in job progress.
+        private void OfflineMapJob_ProgressChanged(object sender, EventArgs e)
+        {
+            // Get the job.
+            GenerateOfflineMapJob job = sender as GenerateOfflineMapJob;
+
+            // Dispatch to the UI thread.
+            Dispatcher.Invoke(() =>
+            {
+                // Show the percent complete and update the progress bar.
+                Percentage.Text = job.Progress > 0 ? job.Progress.ToString() + " %" : string.Empty;
+                ProgressBar.Value = job.Progress;
+            });
+        }
+
+        private void CancelJobButton_Click(object sender, RoutedEventArgs e)
+        {
+            // The user canceled the job.
+            _generateOfflineMapJob.Cancel();
+        }
+
+        #region Authentication
```

Comment:
I agree that it would be better to get that error earlier in the process. Unfortunately, no exception is raised, so there's no way to track whether the action was canceled. Changing how this works would substantially complicate the sample, including the parts currently focused on the overrides/offline map workflow.